### PR TITLE
feat: "Following" button to "Unfollow" on hover

### LINKED ISSUE #1468: Feature: hovering "Following" button should indicate unfollow action
### Type of feature

🍕 Feature

### Current behavior

Currently, when clicking or hovering over the "Following" button on someone's profile, it's not clear what the action will be when clicking it:

Screenshot 2023-08-02 at 9 34 03 AM

Further, this pop up closes immediately on clicking the "Following" button so it's confusing what action was taken. Did clicking that simply cause the menu to collapse? 

Screenshot 2023-08-02 at 9 49 03 AM

Re-opening the menu shows that I've unfollowed the person.

Screenshot 2023-08-02 at 9 49 06 AM

### Suggested solution

I think Twitter/X has a few good, clear solutions for this: 

A) On desktop, when hovering over the following button, it turns red and changes to "Unfollow":

Screenshot 2023-08-02 at 9 43 26 AM

B) on mobile, clicking "Following" pops open a secondary button for unfollowing someone which is also very clear.

IMG_4BF96DD6F6DE-1

---

I'm curious to hear what others prefer in this user flow? I could see either solution above working or coming up with a novel solution. Either way, the UX around unfollowing someone is abit confusing.

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by ThePiyushAggarwal**
## Description

"Following" button changed to "Unfollow" when hovered/focused on other user pages when the user is logged in.

Contributor Profile Header changes
- How? Made use of Tailwind CSS selectors in this. Two `span` tags are used, one of them is hidden based on if it is hovered or not.
- 🌟 Made the dropdown accessible. The button earlier was not getting clicked using keyboard.

Top Contributors changes
- Changed the "following" to "unfollow" when hovered/focused. Accessible with keyboard.

Fixes #1468 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #1468 

## Mobile & Desktop Screenshots/Recordings

|  | Before | After |
| ---- | ---- | ---- |
| Profile Header Menu | image |   <video src="https://github.com/open-sauced/app/assets/27003616/64f8cce3-9ac7-4cc3-abfd-6fb065604173"  /> |
| Highlights Top Contributors Card |  image |  <video src="https://github.com/open-sauced/app/assets/27003616/740988a5-56df-4b64-a14c-9f72a1fbff0d"  /> |

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by OgDev-01 — APPROVED** [review]
Nice!, thanks for turning this around

**Comment by ThePiyushAggarwal** [comment]
> Nice!, thanks for turning this around 🍕

Thank you, I’m glad I could help!

**Review by jpmcb — CHANGES_REQUESTED** [review]
https://github.com/open-sauced/app/assets/23109390/150754ed-bd6e-426f-85a7-5378924c6fed

This behavior is pretty inconsistent: notice in my video how it only changes when it's _directly_ over the button which feels like a sharp edge.

**Comment by bdougie** [comment]
Thanks for jumping on this. We have a following button in other places that we would want to repeat this interaction. i.e. Top Contributors on the highlights page. 

Screen Shot 2023-08-22 at 4 29 58 PM

File: components/molecules/ContributorProfileHeader/contributor-profile-header.tsx [hunk 1]
```diff
@@ -180,9 +180,21 @@ const ContributorProfileHeader = ({
                 {user ? (
                   !isOwner && (
                     <>
+                      {/* Follow, Unfollow button */}
```
[the remark this answers is not in the document]
Comment (ThePiyushAggarwal) [reply components/molecules/ContributorProfileHeader/contributor-profile-header.tsx:183]: Shall I remove this? I have added this just for reference.

**Comment by ThePiyushAggarwal** [comment]
Thanks for reviewing! I will work further on the changes mentioned.

File: components/molecules/ContributorProfileHeader/contributor-profile-header.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ThePiyushAggarwal) [reply components/molecules/ContributorProfileHeader/contributor-profile-header.tsx]: Removed the comment.

**Comment by ThePiyushAggarwal** [comment]
Hi guys! 
The PR is ready for review again. I have made some accessibility improvements as well. Please check out the updated PR description (videos too).

**Review by jpmcb — COMMENTED** [review]
👏🏼 Very nice!!! This is getting closer. I noticed a few other sharp edges:
1. When selecting "Follow" for someone on the main highlights page, the button doesn't update until I click somewhere else. It only changes to "Unfollow". Is that element not being updated properly when a follow occurs?
2. This introduces some spacing issues with the "Following" button on profiles (notice the spacing issue when the button is paired with the "Collaborate" button. It should be centered)


https://github.com/open-sauced/app/assets/23109390/48f56dc5-f155-4d02-b1ca-e6a1ef9fb81a

**Comment by ThePiyushAggarwal** [comment]
Thanks for reviewing! I understand the issues. The first one is happening because the button is still in focus after clicking it. I will work on solutions, make the changes and update the PR.

**Comment by ThePiyushAggarwal** [comment]
Fixed the issues pointed out [above](https://github.com/open-sauced/app/pull/1588#pullrequestreview-1592393292).

### 1. Top Contributors - "unfollow" not getting replaced with "following"
[focus-follow-button-fix.webm](https://github.com/open-sauced/app/assets/27003616/b2ede681-e033-4f59-8e2e-3f43f4bb6565)

### 2. Profile Header - Alignment issue with "Collaborate" button
[follow-button-alignment-fix.webm](https://github.com/open-sauced/app/assets/27003616/61baa256-2074-4663-a4a2-2d00b4d0e5c3)

**Comment by OgDev-01** [comment]
Nicely done @ThePiyushAggarwal 🍕🍕.

**Review by OgDev-01 — APPROVED** [review]

**Review by jpmcb — APPROVED** [review]

**Comment by ThePiyushAggarwal** [comment]
🍕🫡🍕

**Pull request merged**